Antd updates

ChristopherChudzicki opened the pull request:
<!--- # Please remember to use `describe` and `it`in the tests! see https://github.com/flow-typed/flow-typed/blob/master/CONTRIBUTING.md for details. --->

- Links to documentation: https://ant.design/docs/react/introduce
- Link to GitHub or NPM: https://github.com/ant-design/ant-design
- Type of contribution: addition

Other notes: This PR adds a few missing `antd` components, as well as props for some of them.

pascalduez commented on definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js:
```diff
@@ -27,7 +27,25 @@ declare module "antd" {
+  declare export type ButtonProps = {
+    disabled?: boolean,
+    ghost?: boolean,
+    href?: boolean,
```
`string`

pascalduez commented on definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js:
```diff
@@ -27,7 +27,25 @@ declare module "antd" {
+    type?: 'primary' | 'ghost' | 'dashed' | 'danger' | 'default',
+    onClick?: (event?: SyntheticEvent<HTMLButtonElement>) => void,
+    block?: boolean
+  }
```
Missing `target?: string`

pascalduez commented on definitions/npm/antd_v3.x.x/flow_v0.25.x-/test_antd_v3.x.x.js:
```diff
@@ -442,6 +477,13 @@ describe("Tooltip", () => {
+  it("should accept only certain strings for placement prop", () => {
+    // Testing placement prop in particular just to verify sharing between
+    // Popconfirm, Popover, and Tooltip is working.
+    const good = <Popover placement='top' />;
```
copy pasta, must be `Tooltip`

pascalduez commented on definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js:
```diff
@@ -345,7 +363,41 @@ declare module "antd" {
+    cancelText?: string,
+    okText?: string,
+    okType?: 'primary' | 'dashed' | 'ghost' | 'danger' | 'default',
+    title?: 'string' | React$Node,
```
`React$Node` already allows strings

pascalduez commented on definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js:
```diff
@@ -345,7 +363,41 @@ declare module "antd" {
+
+  declare export type PopoverProps = {
+    content?: 'string' | React$Node,
+    title?: 'string' | React$Node,
```
`React$Node` already allows strings

pascalduez commented on definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js:
```diff
@@ -345,7 +363,41 @@ declare module "antd" {
+    mouseEnterDelay?: number,
+    mouseLeaveDelay?: number,
+    overlayClassName?: string,
+    overlayStyle?: Object,
```
`Object` === `any` 

`CSSStyleDeclaration` or `{ +[string]: mixed }`

pascalduez commented on definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js:
```diff
@@ -345,7 +363,41 @@ declare module "antd" {
+    trigger?: 'hover' | 'focus' | 'click' | 'contextMenu',
+    visible?: boolean,
+    onVisibleChange?: (visible?: boolean) => void,
+    align?: Object
```
Same, `Object` === `any`

ChristopherChudzicki commented:
@pascalduez Thanks for the review, and for finding those errors! I did not know about `CSSStyleDeclaration`; that's great.

I think I've addressed all the issues you pointed out. Let me know if there's anything else that should be done.
